HW3/P4/median_filter.cl

```diff
@@ -28,7 +100,20 @@ median_3x3(__global __read_only float *in_values,
     // We've given you median9.h, and included it above, so you can
     // use the median9() function.
 
-
     // Each thread in the valid region (x < w, y < h) should write
     // back its 3x3 neighborhood median.
+
+    // write output
+    if ((y < h) && (x < w)) // stay in bounds
+
+
+        //global version
+        //out_values[y * w + x] = median9(in_values[GetPIX(x-1,y-1,w,h)], in_values[GetPIX(x-1,y,w,h)], in_values[GetPIX(x-1,y+1,w,h)],
+        //                                in_values[GetPIX(x,y-1,w,h)], in_values[GetPIX(x,y,w,h)], in_values[GetPIX(x,y+1,w,h)],
+        //                                in_values[GetPIX(x+1,y-1,w,h)], in_values[GetPIX(x+1,y,w,h)], in_values[GetPIX(x+1,y+1,w,h)]);
+
+        //local version
+        out_values[y * w + x] = median9(buffer[GetPIX(buf_x-1,buf_y-1,buf_w,buf_h)], buffer[GetPIX(buf_x-1,buf_y,buf_w,buf_h)], buffer[GetPIX(buf_x-1,buf_y+1,buf_w,buf_h)],
```

Comment:
You don't need to call GetPIX here, because of the way the buffer was allocated and populated (for example, you'll never get buf-x to be negative or above the buffer's width).